src/QsCompiler/LanguageServer/Program.cs

```diff
@@ -85,6 +85,8 @@ private static int LogAndExit(ReturnCode code, string? logFile = null, string? m
             return (int)code;
         }
 
+        private static string? NotebookReferencesDir => Environment.GetEnvironmentVariable("NOTEBOOK_REFERENCES_DIR");
```

Comment:
Just some bikeshedding on the name - environment variables share a global namespace so a vendor prefix seems like a good idea Maybe `QSHARP_NOTEBOOK_REFERENCES`?